spec/commands/list_spec.rb

```diff
@@ -4,21 +4,70 @@
   before do
     install_gemfile <<-G
       source "file://#{gem_repo1}"
+
       gem "rack"
+      gem "rspec", :group => [:test]
     G
   end
 
   context "with name-only and paths option" do
     it "raises an error" do
       bundle "list --name-only --paths"
+
       expect(out).to eq "The `--name-only` and `--paths` options cannot be used together"
     end
   end
 
+  context "with without and only option" do
+    it "raises an error" do
+      bundle "list --without dev --only test"
+
+      expect(out).to eq "The `--only` and `--without` options cannot be used together"
+    end
+  end
+
+  describe "with without option" do
+    context "when group is present" do
+      it "prints the gems not in the specified group" do
+        bundle! "list --without test"
+
+        expect(out).to include("  * rack (1.0.0)")
+      end
+    end
+
+    context "when group is not found" do
+      it "raises an error" do
+        bundle "list --without random"
+
+        expect(out).to eq "`random` group could not be found."
+      end
+    end
+  end
+
+  describe "with only option" do
+    context "when group is present" do
+      it "prints the gems in the specified group" do
+        bundle! "list --only default"
+
+        expect(out).to include("  * rack (1.0.0)")
```

Comment:
You might also want to ensure that the output does _not_ include gems in other groups.